### ISSUE TITLE
Improve 

### ISSUE THREAD

**Issue by boersmamarcel** [AI4FinTech/FinTorch#47]
- FinTorch version: 0.1.2/0.1.3

### Description
We have the GraphBEAN model which we made semi-supervised. However, we assume that the classes are 1/2/3 where 3 is the unlabelled class which we ignore in the loss function. See:
```
def GraphBEANLossClassifier(
    feature_predictions,
    edge_predictions,
    ground_truth_sampled_data,
    edge,
    node_pred,
    node_ground_truth,
):
    """
    Calculates the total loss for the GraphBEAN model with a classifier.

    Args:
        feature_predictions (Tensor): Predictions of the node features.
        edge_predictions (Tensor): Predictions of the edge features.
        ground_truth_sampled_data (Tensor): Ground truth data.
        edge (Tensor): Edge tensor.
        node_pred (Tensor): Predictions of the node labels.
        node_ground_truth (Tensor): Ground truth node labels.

    Returns:
        Tensor: The total loss for the GraphBEAN model with a classifier.
    """

    # Loss of the feature reconstruction per node type
    loss = GraphBEANLoss(feature_predictions, edge_predictions,
                         ground_truth_sampled_data, edge)

    classification_loss_fn = torch.nn.CrossEntropyLoss()

    # We assume that we have classes 1/2/3 where 3 is unlabbeled. We filter on 3.
    # Define more elegant solution see issue 
    idx = node_ground_truth != 3
    filtered = node_ground_truth[idx]


    class_loss = classification_loss_fn(node_pred[idx, :], filtered)

    weight = 3  # weight > 1, then class loss is weighted more
    # Harmonic mean of feature_loss and total_loss
    harmonic_mean = (1 + weight) * (loss * class_loss) / (weight * loss +
                                                          class_loss)
    total_loss = harmonic_mean

    return total_loss
```

this could be improved and made more robust for other datasets. Moreover, the weight of the harmonic mean is set to 3 which could be a variable.
